src/Mcfedr/DoctrineDelayQueueDriverBundle/Manager/DoctrineDelayQueueManager.php

```diff
@@ -55,7 +55,9 @@ public function put($name, array $arguments = [], array $options = [])
             }
         } elseif (isset($options['delay'])) {
             $jobTime = new \DateTime("+{$options['delay']} seconds", new \DateTimeZone('UTC'));
-        } else {
+        }
+
+        if (!isset($jobTime) || $jobTime < (new \DateTime())->add(new \DateInterval('PT30S'))) {
```

Comment:
I prefer the syntax in the line above `new \DateTime("+30 seconds", new \DateTimeZone('UTC'))` - also notice the use of UTC